### PR TITLE
impl(rest): ignore unknown fields when parsing JSON to proto

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10700)

### CODE REVIEW

File: google/cloud/internal/rest_stub_helpers_test.cc [hunk 1]
```diff
@@ -69,6 +69,33 @@ TEST(RestStubHelpers, RestResponseToProtoErrorInfo) {
...
+    "my_unknown_field":"unknown"
+  }
+)";
+
+TEST(RestStubHelpers, RestResponseToProtoContainsUnknownField) {
+  auto mock_200_response = absl::make_unique<MockRestResponse>();
+  EXPECT_CALL(*mock_200_response, StatusCode()).WillOnce([]() {
+    return HttpStatusCode::kOk;
+  });
+  EXPECT_CALL(std::move(*mock_200_response), ExtractPayload).WillOnce([&] {
```
Comment (devbww) [inline google/cloud/internal/rest_stub_helpers_test.cc:86]: There is another `std::move(*mock_200_response)` below.

File: google/cloud/internal/rest_stub_helpers_test.cc [hunk 2]
```diff
@@ -69,6 +69,33 @@ TEST(RestStubHelpers, RestResponseToProtoErrorInfo) {
...
+    return MakeMockHttpPayloadSuccess(
+        std::string(kJsonRolePayloadWithUnknownField));
+  });
+
+  google::iam::admin::v1::Role role;
+  auto status = RestResponseToProto(role, std::move(*mock_200_response));
+  EXPECT_THAT(status, IsOk());
```
Comment (devbww) [inline google/cloud/internal/rest_stub_helpers_test.cc:93]: `ASSERT_THAT(...)`?  All the following expectations seem useless without OK.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10700)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @devbww)

___
*[`google/cloud/internal/rest_stub_helpers_test.cc` line 86 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10700#-NN4h_QMc9WGDn-1hURI-r1-86:-NN4xE_WE3JN8XJJZ1D0:bd0u6i5) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/1e8a39661e452f376c129d369f4705eab7845ad8/google/cloud/internal/rest_stub_helpers_test.cc#L86)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

There is another `std::move(*mock_200_response)` below.
</blockquote></details>

Having the `std::move` in the `EXPECT_CALL` is the only way I know of to mock an rvalue-ref qualified member function.
___
*[`google/cloud/internal/rest_stub_helpers_test.cc` line 93 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10700#-NN4h_QMc9WGDn-1hURI-r1-93:-NN4xDJ00d1OeJEi6YZI:bt2vr1b) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/1e8a39661e452f376c129d369f4705eab7845ad8/google/cloud/internal/rest_stub_helpers_test.cc#L93)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

`ASSERT_THAT(...)`?  All the following expectations seem useless without OK.
</blockquote></details>

Fixed.

**Review by devbww — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10700)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @scotthart)

___
*[`google/cloud/internal/rest_stub_helpers_test.cc` line 86 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10700#-NN4h_QMc9WGDn-1hURI-r1-86:-NN5TtBmCoGNHKOWkkPn:b8a4ip3) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/1e8a39661e452f376c129d369f4705eab7845ad8/google/cloud/internal/rest_stub_helpers_test.cc#L86)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

Having the `std::move` in the `EXPECT_CALL` is the only way I know of to mock an rvalue-ref qualified member function.
</blockquote></details>

Right you are.  It seems things are carefully crafted to not consume the mock object, despite the `&&` call.  Thanks.

**Pull request merged**